python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -234,3 +243,29 @@ def evaluate_dask(ir: IR) -> DataFrame:
 
     graph, key = task_graph(ir, partition_info)
     return get(graph, key)
+
+
+def _concat(dfs: Sequence[DataFrame]) -> DataFrame:
+    # Concatenate a sequence of DataFrames vertically
+    return Union.do_evaluate(None, *dfs)
+
+
+##
+## DataFrameScan
+##
+
+
+@lower_ir_node.register(DataFrameScan)
+def _(
+    ir: DataFrameScan, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
+    from cudf_polars.experimental.io import ParDataFrameScan
+
+    new_node = ParDataFrameScan(
+        ir.schema,
+        ir.df,
+        ir.projection,
+        ir.predicate,
+        ir.config_options,
+    )
```

Comment:
*question*: Do we need a new node for this one? Or can we do it here?

Specifically, IIUC this could be (with appropriate lowering for `Union`):

```
rows_per_partition = ir.config_options.get(
    "parallel_options", {}
).get("num_rows_threshold", 1_000_000)

nrows = max(ir.df.shape()[0], 1)
npartitions = math.ceil(nrows / rows_per_partition)
length = math.ceil(nrows / npartitions)

slices = [
    DataFrameScan(
        ir.schema, 
        ir.df.slice(offset, length), 
        ir.projection,
        ir.predicate,
        ir.config_options,
   )
   for offset in range(0, nrows, length)
]
new_node = Union(ir.schema, *slices)
return new_node, {new_node: PartitionInfo(count=npartitions)}
```

If you want a different node for `Partitioned` as opposed to using `Union` (which semantically takes many partitions and produces a single one), I would not be opposed to introducing that instead.

But I want to try and avoid, if possible, replicating lots of nodes as `ParFoo` for some "serial" node `Foo` if we can express this with a few "primitives".